docs/guide/middleware.md

```diff
@@ -0,0 +1,90 @@
+Middleware
+==========
+
+Middleware will be closure or class. Class must implemented by yii\web\MiddlewareInterface
+
+## Global
+
+Global middleware for all routes can add in application config:
+
+```php
+        [
+            'middleware' => \app\middleware\CopyrightMiddleware::class,
+            'priority' => 10,
+        ],        
+        function (\yii\web\Request $request, \yii\web\Response $response) {
+            $response->addHeader("heaader", "custom");
+        }
+```
+
+```php
+namespace app\middleware;
+
+use yii\web\MiddlewareInterface;
+use yii\web\Request;
+use yii\web\Response;
+
+class CopyrightMiddleware implements MiddlewareInterface
+{
+    public function process(Request $request, Response $response)
+    {
+        $response->addHeader('copyright', 'Yiisoft');
+    }
+}
+```
+
+## Router
+
+```php
+'urlManager' => [
+    'enablePrettyUrl' => true,
+    'showScriptName' => false,
+    'rules' => [
+        [
+            'pattern' => '/',
+            'route' => 'site/index',
+            'middleware' => [
+                [
+                    'middleware' => function ($request, \yii\web\Response $response) {
```

Comment:
Config syntax looks a bit weird with `middleware.middleware`

Reply:
I agree. But i has not decided yet how to call it

Reply:
fixed